### ISSUE TITLE
Nonexistent remote HEAD ref

### ISSUE THREAD

**Issue by k15z** [src-d/go-git#645]
Running `git clone https://github.com/lodyagin/bitcoinj` produces a warning, but you can list branches and check out the correct one.

> warning: remote HEAD refers to nonexistent ref, unable to checkout.

How would you do this in `go-git`? The below code results in `plumbing.ErrReferenceNotFound`.

	repository, err := git.Clone(memory.NewStorage(), memfs.New(), &git.CloneOptions{
		URL: "https://github.com/lodyagin/bitcoinj",
	})

It seems like this is currently not possible?

**Comment by mcuadros** [comment]
Yep, this is not possible right now. Maybe an error can be returned but also the repository, more lake a warning instead of fatala.
